nucliadb_vectors2/src/vectors/data_point/mod.rs

```diff
@@ -122,6 +122,14 @@ impl<'a, Dlog: DeleteLog> Retriever<'a, Dlog> {
 }
 
 impl<'a, Dlog: DeleteLog> DataRetriever for Retriever<'a, Dlog> {
+    fn get_vector(&self, x @ Address(addr): Address) -> &[u8] {
+        if addr == key_value::get_no_elems(self.nodes) {
+            self.temp
+        } else {
+            let x = self.find_node(x);
+            Node::vector(x)
+        }
+    }
```

Comment:
Maybe we can remove the destructing pattern `x @ Address(addr)` parameter and implement on `Address` type a `inner` method to improve maintainability and readability ?
Because, even if the destructing pattern is super useful time to time, this is rarely the case in function parameter (the code will break in a instant if we modify a bit the `Address` type).

Reply:
Address is a internal type that must never be exposed, as well as the methods that use such type. For this reason I dont want to implement a inner method nor Deref trait, if Address changes, the code that uses it should complain and carefully modified. 
The same thing is taken in the Rust compiler when dealing with the MIR, matching variants of the IR must be  exhaustive for safety reasons.

Reply:
Ok, sounds fair (even if I truly believe that internal types should be treated as exposed one)

Reply:
Lets revisit this when I get back, maybe there is an approach that gives both flexibility and security.